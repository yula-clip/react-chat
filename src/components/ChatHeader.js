import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import MoreIcon from '@material-ui/icons/MoreVert';
import IconButton from '@material-ui/core/IconButton';
import AccountCircle from '@material-ui/icons/AccountCircle';
import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';

const styles = theme =>({
  appBar: {
    position: 'fixed',
    width: `calc(100% - 320px)`,
  },
  grow: {
    flexGrow: 1,
  },
  avatar: {
    marginRight: '16px',
  },
});

const ChatHeader = ({classes}) => (
  <AppBar color = 'primary' className={classes.appBar}>
          <Toolbar>
            <Avatar className={classes.avatar}>H</Avatar>
            <Typography variant="h6" color="inherit" noWrap>
              DogeCodes React Chat
            </Typography>
            <IconButton 
             color="inherit">
              <MoreIcon />
            </IconButton>
            <div className={classes.grow} />
            <div>
                <IconButton
                  aria-haspopup="true"
                  color="inherit"
                >
                  <AccountCircle />
                </IconButton>
            </div>
          </Toolbar>
        </AppBar>
);

export default withStyles(styles)(ChatHeader);
