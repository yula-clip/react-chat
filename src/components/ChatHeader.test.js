/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ChatHeader from "./ChatHeader";

jest.mock("./Avatar", () => () => "Avatar");
jest.mock("./ChatMenu", () => () => "ChatMenu");
jest.mock("./UserMenu", () => () => "UserMenu");

const mockProps = {
  activeUser: {
    isMember: true,
    isCreator: true,
    isChatMember: true
  },
  activeChat: {
    _id: "12345",
    title: "My Chat"
  },
  messages: [
    {
      chatId: "12345",
      content: "content",
      sender: {},
      createdAt: "2018-03-16T10:53:23.200Z"
    }
  ],
  isConnected: true,
  logout: jest.fn(),
  leaveChat: jest.fn(),
  deleteChat: jest.fn(),
  editUser: jest.fn()
};

describe("<ChatHeader />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ChatHeader {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<ChatHeader {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders without active chat", () => {
    const tree = renderer
      .create(<ChatHeader {...mockProps} activeChat={null} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
