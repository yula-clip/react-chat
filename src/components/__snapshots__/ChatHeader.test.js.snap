// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> renders correctly 1`] = `
<header
  className="MuiPaper-root-91 MuiPaper-elevation4-97 MuiAppBar-root-82 MuiAppBar-positionFixed-83 MuiAppBar-colorPrimary-89 mui-fixed ChatHeader-appBar-79"
>
  <div
    className="MuiToolbar-root-118 MuiToolbar-regular-120 MuiToolbar-gutters-119"
  >
    Avatar
    <h6
      className="MuiTypography-root-122 MuiTypography-h6-139 MuiTypography-colorInherit-151 MuiTypography-noWrap-148"
    >
      My Chat
      ChatMenu
    </h6>
    <div
      className="ChatHeader-grow-80"
    />
    <div>
      UserMenu
    </div>
  </div>
</header>
`;

exports[`<ChatHeader /> renders without active chat 1`] = `
<header
  className="MuiPaper-root-91 MuiPaper-elevation4-97 MuiAppBar-root-82 MuiAppBar-positionFixed-83 MuiAppBar-colorPrimary-89 mui-fixed ChatHeader-appBar-79"
>
  <div
    className="MuiToolbar-root-118 MuiToolbar-regular-120 MuiToolbar-gutters-119"
  >
    <h6
      className="MuiTypography-root-122 MuiTypography-h6-139 MuiTypography-colorInherit-151 MuiTypography-noWrap-148"
    >
      DogeCodes React Chat
    </h6>
    <div
      className="ChatHeader-grow-80"
    />
    <div>
      UserMenu
    </div>
  </div>
</header>
`;
